Add mbed-os-example-pelion

adbridge opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
Add pelion example to the examples.json list. Primarily so that this example can be updated automatically with a pull request when updating Mbed OS version.
Set to compile in the CI but not export or test.


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

adbridge committed:
Add mbed-os-example-pelion

Primarily so that this example can be updated automatically with
a pull request when updating Mbed OS version.
Set to compile in the CI but not export or test.

teetak01 commented on tools/test/examples/examples.json:
```diff
@@ -463,6 +463,22 @@
+        "compile" : true,
+        "export": false,
+        "test" : false,
+        "baud_rate": 9600,
```
baud_rate is wrong. Should be 115200

adbridge committed:
Update baud rate of pelion example

Should be 115200

adbridge committed:
Update mbed-os-example-pelion for supported targets

According to mbed_app.json in the example itself only these specific targets are listed.
"K64F", "K66F", "NUCLEO_F429ZI", "UBLOX_EVK_ODIN_W2", "NUCLEO_F411RE", "DISCO_L475VG_IOT01A"
